deployment-aws/terraform/network.tf

```diff
@@ -151,3 +158,150 @@ resource "aws_security_group" "api_server" {
+  description = "Allow everything out, nothing in"
+  vpc_id      = aws_vpc.main.id
+  egress {
+    description      = "Allow all egress"
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+}
+
+# For ECS instances launched into the private subnet, they can communicate
+# with the container registry via this vpc endpoint
+resource "aws_vpc_endpoint" "ecr_dkr" {
+  vpc_id            = aws_vpc.main.id
+  service_name      = "com.amazonaws.${data.aws_region.current.name}.ecr.dkr"
+  vpc_endpoint_type = "Interface"
+  subnet_ids        =  [aws_subnet.private_a.id, aws_subnet.private_b.id]
+  security_group_ids = [
+    aws_security_group.endpoint_security_group.id
+  ]
+  policy =<<POLICY
+  {
+    "Statement": [
+      {
+        "Sid": "AllowPull",
+        "Principal": {
+          "AWS": "${aws_iam_role.ecs_execution_role.arn}"
+        },
+        "Action": [
+          "ecr:BatchGetImage",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetAuthorizationToken"
+        ],
+        "Effect": "Allow",
+        "Resource": "*"
+      }
+    ]
+  }
+  POLICY
+  private_dns_enabled = true
+}
+
+# For ECS instances launched into the private subnet, they can communicate
+# with the container registry via this vpc endpoint
+resource "aws_vpc_endpoint" "ecr_api" {
+  vpc_id            = aws_vpc.main.id
+  service_name      = "com.amazonaws.${data.aws_region.current.name}.ecr.api"
+  vpc_endpoint_type = "Interface"
+  subnet_ids        =  [aws_subnet.private_a.id, aws_subnet.private_b.id]
+  security_group_ids = [
+    aws_security_group.endpoint_security_group.id
+  ]
+  policy =<<POLICY
+  {
+    "Statement": [
+      {
+        "Sid": "AllowPull",
+        "Principal": {
+          "AWS": "${aws_iam_role.ecs_execution_role.arn}"
+        },
+        "Action": [
+          "ecr:BatchGetImage",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetAuthorizationToken"
+        ],
+        "Effect": "Allow",
+        "Resource": "*"
+      }
+    ]
+  }
+  POLICY
+  private_dns_enabled = true
+}
+
+
+# Permits the ECS instances in the private subnet to communicate with 
+# logging mechanism
+resource "aws_vpc_endpoint" "logs_endpoint" {
+  vpc_id            = aws_vpc.main.id
+  service_name      = "com.amazonaws.${data.aws_region.current.name}.logs"
+  vpc_endpoint_type = "Interface"
+  subnet_ids        =  [aws_subnet.private_a.id, aws_subnet.private_b.id]
+  security_group_ids = [
+    aws_security_group.endpoint_security_group.id
+  ]
+  private_dns_enabled = true
+}
+
+# Although we don't explicitly create a route table for the 
+# private subnets, it is created automatically and associated
+# with both
+# data "aws_route_table" "private_route_table" {
+#   vpc_id    = aws_vpc.main.id
+#   subnet_id = aws_subnet.private_a.id
+# }
+
+resource "aws_route_table" "private_route_table" {
```

Comment:
No need to include the resource type in the name:
```
resource "aws_route_table" "private"
```
Then you can refer to it without redundancy:
```
aws_route_table.private
```